enigma-swing/src/main/java/cuchaz/enigma/gui/panels/right/MessagesPanel.java

```diff
@@ -0,0 +1,85 @@
+package cuchaz.enigma.gui.panels.right;
+
+import cuchaz.enigma.gui.Gui;
+import cuchaz.enigma.network.packet.MessageC2SPacket;
+import cuchaz.enigma.utils.I18n;
+
+import javax.swing.AbstractAction;
+import javax.swing.JButton;
+import javax.swing.JPanel;
+import javax.swing.JScrollPane;
+import javax.swing.JTextField;
+import java.awt.BorderLayout;
+import java.awt.event.ActionEvent;
+
+public class MessagesPanel extends MultiplayerOnlyRightPanel {
+    private final Gui gui;
+    private final JScrollPane messageScrollPane;
+    private final JTextField pendingMessageBox;
+    private final JButton sendPendingMessageButton;
+    private final JPanel chatPanel;
+
+    public MessagesPanel(Gui gui) {
+        super(gui);
+        this.gui = gui;
```

Comment:
`gui` should be a protected field of `RightPanel`, instead of a private one for `CallsTree`, `StructurePanel` and `MessagesPanel`, which also shadows a private one on `MultiplayerOnlyRightPanel`